scrnaseq.Rmd

```diff
@@ -385,25 +432,34 @@ for (i in seq(nrow(datasets))) {
   
   # Read 10X or smartseq2
   if (type == "10x") {
-    
     # Read 10X sparse matrix into a Seurat object
-    sc[[name]] = ReadSparseMatrix(path, 
+    sc = c(sc, ReadSparseMatrix(path, 
                                   project=name, 
                                   row_name_column=1, 
-                                  convert_row_names=ensembl_to_seurat_rowname)
+                                  convert_row_names=ensembl_to_seurat_rowname))
     
   } else if (type == "smartseq2") {
-    
     # Read counts table into a Seurat object
+    # Note: If a dataset contains multiple samples, the sample names could clash with sample names of other datasets. This will be taken care of next.
     sc = c(sc, ReadCountsTable(path, project=name, row_name_column=1, convert_row_names=ensembl_to_seurat_rowname, parse_plate_information=TRUE, return_samples_as_datasets=TRUE))
   } 
 }
 
-# Make cell names unique
```

Comment:
Good question. I think that I do not wanted to automatically append a `-1`, `-2` etc to the cell names since it might not be neccessary. On the other hand, for Loupe it is neccessary.

Reply:
Ok, as I am a little lost here, lets discuss this via Zoom later.